src/list.lisp

```diff
@@ -475,3 +475,63 @@
 (defun mapcon (function list &rest more-lists)
   "Apply FUNCTION to successive CDRs of lists. Return NCONC of results."
   (map1 function (cons list more-lists) :nconc nil))
+
+
+;;; set-difference
+(defun set-difference (list1 list2 &key key (test #'eq))
+    (cond (list2
+           (let ((result '()))
+               (dolist (it list1)
+                   (when (not (member it list2 :key key :test test))
+                       (push it result)))
+               result))
+          (t list1)))
+
+;;; makeset
+(defun makeset (lst &key (test #'eq))
+    (prog ((result)
+           (seq lst))
+     feed
+       (when (null seq) (return (reverse result)))
+       (if (not (member (car seq) result :test test))
+           (setq result (cons (car seq) result)))
+       (setq seq (cdr seq))
+       (go feed)))
+
+;;; union
+(defun union (list1 list2 &key key (test #'eq))
+    (cond ((and list1 list2)
+           (let ((result (makeset list2 :test #'equal)))
+               (dolist (it (makeset list1 :test #'equal))
+                   (when (not (member it list2 :key key :test test))
+                       (push it result)))
+               result))
+          (list1)
+          (list2)))
+
+
+;;; sort
+;;; https://en.wikipedia.org/wiki/Selection_sort
```

Comment:
Any reason to use selection sort?  If so, for example, to keep it simple, please add it to the comment of the function. Although it does not look too simple to be honest.

Have you checked what algorithm other implementations are using?

Reply:
It would be nice if you could check for non-lists and give a good error as well, so users know we are aware it is not implemented. :-)

Reply:
Usually i use (#j:sort). But in last year, when I considered the price of Closette integration, I decided that the function sort  would be on the lisp.

The selection criteria were simple:
- do not used additional memory
- without recursion
- simple pseudocode and fast implementation

Selection space - selection, merge, insertion. Why then chose seletion - I do not remember. Did and forgot.

Now, no desire to rewrite code. Unstable algoritm? I believe that it is to may be indicate in the release note. However, if there is a other ready-made algorithm, I am ready to accept it. Our colleagues and distributors can also offer their own implementation. If they deem it necessary.

A good question :+1:  Is that it took more time to explain than implement a function. Considering that in two last years, no one has opened issue about  the absence of a sort function, I consider the price of the question to be minimal. Specifically, I did'nt check the release of the sort function   in the CCL/SBCL. It is possible that now I would do the implementation of bubble sort.  In the release of CLOS, I only need the sort function in one place. Higly likly,  speed bubble sorting will there is enough. I'll think about it.

Check non-list sequence will be. Thats my fault.

Reply:
It is good to leave it as it is. Just add a small comment explaining why. You can copy-paste parts of the comms t above.  Good explanation :+1:

If somebody needs a better sort, they can do it later.